Sources/CvPlayer.cpp

```diff
@@ -2619,10 +2619,9 @@ void CvPlayer::acquireCity(CvCity* pOldCity, bool bConquest, bool bTrade, bool b
 
 	if (bTrade)
 	{
-		foreach_(CvPlot* pLoopPlot, CvPlot::rect(pCityPlot->getX(), pCityPlot->getY(), 1, 1))
-		{
-			pLoopPlot->setCulture(eOldOwner, 0, false, false);
-		}
+		algo::for_each(pCityPlot->rect(1, 1),
+			bind(CvPlot::setCulture, _1, eOldOwner, 0, false, false)
+		);
```

Comment:
this means that only the plot that city is on gets culture swapped? 
setCulture should be investigated further so that culture around the city get atleast partially flipped to conquering civ

Reply:
There is pinned message in cpp channel. I couldn't explain better then bill or google.